tests/plugins/chart/basic_chart.test.ts

```diff
@@ -1446,6 +1448,216 @@ describe("Chart design configuration", () => {
+
+    updateChart(aggregatedModel, "42", { aggregated: true });
+    chart = (aggregatedModel.getters.getChartRuntime("42") as BarChartRuntime).chartJsConfig;
+    expect(chart.data!.datasets![0].data).toEqual([24, 26, 28, 30]);
+    expect(chart.data!.labels).toEqual(["P1", "P2", "P3", "P4"]);
+
+    updateChart(aggregatedModel, "42", { type: "line" });
+    chart = (aggregatedModel.getters.getChartRuntime("42") as BarChartRuntime).chartJsConfig;
+    expect(chart.data!.datasets![0].data).toEqual([24, 26, 28, 30]);
+    expect(chart.data!.labels).toEqual(["P1", "P2", "P3", "P4"]);
+
+    updateChart(aggregatedModel, "42", { aggregated: false });
+    chart = (aggregatedModel.getters.getChartRuntime("42") as BarChartRuntime).chartJsConfig;
+    expect(chart.data!.datasets![0].data).toEqual([10, 11, 12, 13, 14, 15, 16, 17]);
+    expect(chart.data!.labels).toEqual(["P1", "P2", "P3", "P4", "P1", "P2", "P3", "P4"]);
+
+    updateChart(aggregatedModel, "42", { type: "pie" });
+    chart = (aggregatedModel.getters.getChartRuntime("42") as BarChartRuntime).chartJsConfig;
+    expect(chart.data!.datasets![0].data).toEqual([10, 11, 12, 13, 14, 15, 16, 17]);
+    expect(chart.data!.labels).toEqual(["P1", "P2", "P3", "P4", "P1", "P2", "P3", "P4"]);
+
+    updateChart(aggregatedModel, "42", { aggregated: true });
+    chart = (aggregatedModel.getters.getChartRuntime("42") as BarChartRuntime).chartJsConfig;
+    expect(chart.data!.datasets![0].data).toEqual([24, 26, 28, 30]);
+    expect(chart.data!.labels).toEqual(["P1", "P2", "P3", "P4"]);
+  });
+
+  test("One dataset: some (not all) values are lost for a label", () => {
+    createChart(aggregatedModel, aggregatedChart, "42");
+    let chart = (aggregatedModel.getters.getChartRuntime("42") as BarChartRuntime).chartJsConfig;
+    expect(chart.data!.datasets![0].data).toEqual([10, 11, 12, 13, 14, 15, 16, 17]);
+    expect(chart.data!.labels).toEqual(["P1", "P2", "P3", "P4", "P1", "P2", "P3", "P4"]);
+
+    updateChart(aggregatedModel, "42", { aggregated: true });
+    chart = (aggregatedModel.getters.getChartRuntime("42") as BarChartRuntime).chartJsConfig;
+    expect(chart.data!.datasets![0].data).toEqual([24, 26, 28, 30]);
+    expect(chart.data!.labels).toEqual(["P1", "P2", "P3", "P4"]);
+
+    setCellContent(aggregatedModel, "B3", "");
+    setCellContent(aggregatedModel, "B6", "");
+    chart = (aggregatedModel.getters.getChartRuntime("42") as BarChartRuntime).chartJsConfig;
+    expect(chart.data!.datasets![0].data).toEqual([10, 15, 28, 30]);
+    expect(chart.data!.labels).toEqual(["P1", "P2", "P3", "P4"]);
+
+    updateChart(aggregatedModel, "42", { aggregated: false });
+    chart = (aggregatedModel.getters.getChartRuntime("42") as BarChartRuntime).chartJsConfig;
+    expect(chart.data!.datasets![0].data).toEqual([10, undefined, 12, 13, undefined, 15, 16, 17]);
+    expect(chart.data!.labels).toEqual(["P1", "P2", "P3", "P4", "P1", "P2", "P3", "P4"]);
```

Comment:
this is not related to aggregate and should be tested else where. (It is probably already tested)